### PR TITLE
Add loading step on LIB flow

### LINKED ISSUE #66396: SignUp: Add loading step
### Details

After submitting the form there should be a loading step, according to this [design spec](https://www.figma.com/file/BcUdhLlhAp7UHtQHNTI4zA/Tailored-Onboarding-%26-Launchpad?node-id=1865%3A18108)

### Checklist

_No response_

### Related

_No response_

### PR DESCRIPTION
**Pull request by renancarvalho**
#### Proposed Changes

After the link-in-bio setup step, when submitting the form, the user should see the loading step that is basically a loader while the data is being handled/submitted.

It should look like the [designs](https://www.figma.com/file/BcUdhLlhAp7UHtQHNTI4zA/Tailored-Onboarding-%26-Launchpad?node-id=1865%3A18108) besides the background that is not the one on the design, it will be changed later.

image

#### Testing Instructions

 - Run yarn start from the calypso root folder
 - Navigate to http://calypso.localhost:3000/setup/linkInBioSetup?flow=link-in-bio&siteSlug=YOUR-SITE.wordpress.com
 - Fill the form and click on continue
 - You should see the loader as the described above.
 ( It will be fast given we have little data and the whole action happens quite fast )
- After that you should be redirect to temporary page, `http://calypso.localhost:3000/page/YOUR-SITE.wordpress.com/home`. ( will be changed later )

Related to #66396
Fixes #66396

### CODE REVIEW

**Review by escapemanuele — CHANGES_REQUESTED** [review]
Hey @renancarvalho, thanks for this! 👍 
As in the screenshot, to me the content appears not centered, happens to you too?

image

**Comment by renancarvalho** [comment]
Hi @escapemanuele, Thank you for checking it, indeed it is a bit misaligned, pushing the fix soon.
I also noticed that the print you provided is not up to date with the correct text and progress bar.

it should look like these 
image

image


Thank you!

File: client/landing/stepper/declarative-flow/internals/steps-repository/link-in-bio-setup/index.tsx [hunk 1]
```diff
@@ -63,26 +64,38 @@ const LinkInBioSetup: Step = function LinkInBioSetup( { navigation } ) {
...
-			if ( selectedFile ) {
-				try {
-					await setSiteLogo( selectedFile );
-				} catch ( _error ) {
-					// communicate the error to the user
+
+		setPendingAction( async () => {
+			if ( site ) {
+				setProgress( 0 );
+				setProgressTitle( __( 'Turning on the lights' ) );
```
Comment (arcangelini) [inline client/landing/stepper/declarative-flow/internals/steps-repository/link-in-bio-setup/index.tsx]: I think this is supposed to cycle through a bunch of sayings like in this hook ⇢ https://github.com/Automattic/wp-calypso/blob/trunk/client/landing/stepper/declarative-flow/internals/steps-repository/processing-step/hooks/use-processing-loading-messages.ts

We could move that hook to the main /hook folder and make it functional for both steps and potentially more in the future?

**Comment by heavyweight** [comment]
Screenshot 2022-08-10 at 12 27 03 1

Are we missing the image background in the loading step?

**Comment by renancarvalho** [comment]
@heavyweight I appreciate your review.
Yes, we will use this background color for now as @alshakero suggested.

File: client/landing/stepper/declarative-flow/internals/steps-repository/intro/styles.scss [hunk 2]
```diff
@@ -5,6 +5,29 @@ $design-button-primary-color: var( --color-primary );
...
+.link-in-bio {
+	&.processing {
+		background: rgb( 162, 218, 254 );
+		position: fixed;
+		width: 100%;
+		height: 100%;
+
+		.step-container__content {
+		    display: flex;
+			justify-content: center;
```
Comment (arcangelini) [inline client/landing/stepper/declarative-flow/internals/steps-repository/intro/styles.scss:16]: Based on the Figma and your screenshots, I think this is supposed to be aligned to the left with the flow title?

I addressed the flow title in this PR #66416 

Maybe we can merge that first and you can rebase to get the flow title in there?

**Review by arcangelini — CHANGES_REQUESTED** [review]
Just two things I noticed. Besides that this looks good!

File: client/landing/stepper/declarative-flow/internals/steps-repository/intro/styles.scss (hunk 2 above)
Comment (renancarvalho) [reply client/landing/stepper/declarative-flow/internals/steps-repository/intro/styles.scss:16]: Sounds good to merge that one 1st and then rebase.
Thank you for letting me know about it! 👍

**Comment by renancarvalho** [comment]
Not relevant anymore

**Pull request closed without merging**